iosMath/render/internal/MTTypesetter.m

```diff
@@ -849,14 +857,36 @@ - (void) createDisplayAtoms:(NSArray*) preprocessed
     }
 }
 
+
 - (MTCTLineDisplay*) addDisplayLine
 {
     // add the font
     [_currentLine addAttribute:(NSString *)kCTFontAttributeName value:(__bridge id)(_styleFont.ctFont) range:NSMakeRange(0, _currentLine.length)];
     /*NSAssert(_currentLineIndexRange.length == numCodePoints(_currentLine.string),
      @"The length of the current line: %@ does not match the length of the range (%d, %d)",
      _currentLine, _currentLineIndexRange.location, _currentLineIndexRange.length);*/
-    
+
+    // add the color
+    NSString *immutLine = [_currentLine string];
```

Comment:
You don't need to do any of this. If you do the fusing above correctly, you'll be guaranteed that entire displayLine has the same color. You can just set the textColor for the line.